### PR TITLE
Get Demo page working

### PR DESCRIPTION
**Pull request by pbrisbin**
These changes get comments actually working on the Demo page. Along with
https://github.com/thoughtbot/maitre-d/pull/103, the blog is unaffected.

### CODE REVIEW

File: templates/embed.lucius [hunk 1]
```diff
@@ -7,8 +7,9 @@
 
 .carnival-comment-indicator {
   cursor: pointer;
-  position: absolute;
-  right: -30px;
+  float: right;
+  position: relative;
+  right: -15px;
```
Comment (pbrisbin) [inline templates/embed.lucius]: This is the only way I could find to do this without relying on the specifics of exactly where and how wide the user's `article` element is. It does cause some minor text-re-flow when the indicators get added to the DOM though. Unfortunately, I don't see a way to address this without re-working the entire approach (which I think we should do eventually).

File: templates/demo.hamlet [hunk 2]
```diff
@@ -2,19 +2,82 @@
...
+    dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut
+    rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et
+    molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente
+    delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut
+    perferendis doloribus asperiores repellat.
 
   <p>
-    View the page source to see how simple it was to install.
 
-  <pre>
```
Comment (jferris) [inline templates/demo.hamlet]: Do we not want a code example?

File: templates/embed/Article.coffee [hunk 3]
```diff
@@ -39,3 +41,18 @@ class Article
...
+      @element.style.position = 'relative'
+      @element.style.left = '-150px'
+
+  restoreArticle: =>
+    if @wideScreen()
+      @element.style.position = @prevPosition
+      @element.style.left = @prevLeft
+
+  wideScreen: =>
+    not window.matchMedia('only screen and (max-width: 640px)').matches
```
Comment (pbrisbin) [inline templates/embed/Article.coffee]: http://caniuse.com/#feat=matchmedia

File: templates/demo.hamlet (hunk 2 above)
Comment (pbrisbin) [inline templates/demo.hamlet]: This whole thing'll go away when we find a real page to add comments to.

**Comment by pbrisbin** [comment]
With the article `position: relative`, I was able to get things working with only the two minor changes that we were doing manually over on the blog anyway. This is good because it means I don't have to do a synchronized deploy.

Removing the `float: right` on the form was a preference thing, I think it looks better without it.

File: templates/embed/Article.coffee [hunk 4]
```diff
@@ -39,3 +42,15 @@ class Article
     container.className = 'carnival'
     @element.insertBefore(container, @element.firstChild)
     container
+
+  shiftArticle: =>
+    if @wideScreen()
+      @prevLeft = @element.style.left
+      @element.style.left = '-150px'
```
Comment (jferris) [inline templates/embed/Article.coffee]: Should we subtract 150 from the previous left?

Comment (pbrisbin) [inline templates/embed/Article.coffee]: Sure. I wonder if `@element.style.left` could return `"200em"`...

Comment (jferris) [inline templates/embed/Article.coffee]: This is the Internet! Anything can happen!

**Comment by pbrisbin** [comment]
@jferris is this good to go?

File: test/Handler/CommentsSpec.hs [hunk 5]
```diff
@@ -43,7 +42,7 @@ spec = withApp $ do
 
             get $ CommentsR siteId
 
-            valueEquals =<< commentsResponse siteId u [c2, c3, c1]
+            valueEquals =<< commentsResponse u [c2, c3, c1]
```
Comment (pbrisbin) [inline test/Handler/CommentsSpec.hs]: I must've botched this when I rebased the feed-fix over the comment-order fix.

File: test/Handler/CommentsSpec.hs [hunk 6]
```diff
@@ -6,7 +6,6 @@ module Handler.CommentsSpec
 import TestImport
 import Model.UserComment
 import Data.Aeson (Value, (.=), encode, object)
-import Data.Time (getCurrentTime)
```
Comment (pbrisbin) [inline test/Handler/CommentsSpec.hs]: ClassyPrelude seems to have a lot more out of the box than Prelude.

**Pull request merged**